spec/unit/defaults_spec.rb

```diff
@@ -190,78 +190,21 @@
     end
   end
 
-  describe "the call hook for the cadir setting", unless: Puppet::Util::Platform.windows? do
-    it 'does not warn when the cadir is outside the puppet ssldir' do
-      FileUtils.mkdir_p(Puppet[:confdir])
-      # This is fun; in order to get the confdir setting from the spec_helper to
-      # not be deleted by the #unsafe_clear in Puppet::Settings#parse_config,
-      # we can code it into the confdir itself, and thus we can avoid getting
-      # that setting wiped out during Puppet.initialize_settings.
-      File.write(File.join(Puppet[:confdir], 'puppet.conf'),
-                 "cadir = /my/cool/path/for/my/cadir\n
-                  confdir = #{Puppet[:confdir]}")
-      expect(Puppet).to_not receive(:log_ca_migration_warning)
-      Puppet.initialize_settings
-      expect(Puppet[:cadir]).to eq("/my/cool/path/for/my/cadir")
-    end
-
-    it 'does warn when the cadir is inside the puppet ssldir' do
-      FileUtils.mkdir_p(Puppet[:confdir])
-      cadir_location = File.join(Puppet[:ssldir], 'still_inside_ssldir')
-      # This is fun; in order to get the confdir setting from the spec_helper to
-      # not be deleted by the #unsafe_clear in Puppet::Settings#parse_config,
-      # we can code it into the confdir itself, and thus we can avoid getting
-      # that setting wiped out during Puppet.initialize_settings.
-      File.write(File.join(Puppet[:confdir], 'puppet.conf'),
-                 "cadir = #{cadir_location}\n
-                  confdir = #{Puppet[:confdir]}")
-      expect(Puppet).to receive(:log_ca_migration_warning).twice
-      Puppet.initialize_settings
-      expect(Puppet[:cadir]).to eq(cadir_location)
-      end
-  end
-
   describe "the default cadir", :unless => Puppet::Util::Platform.windows?  do
-    it 'defaults to the puppetserver confdir' do
+    it 'defaults to the puppetserver confdir when no cadir is found' do
       Puppet.initialize_settings
       expect(Puppet[:cadir]).to eq('/etc/puppetlabs/puppetserver/ca')
     end
 
-    it 'returns an empty string for Windows platforms', :if => Puppet::Util::Platform.windows? do
-      Puppet.initialize_settings
-      expect(Puppet[:cadir]).to eq("")
-    end
-  end
-
-  describe '#default_cadir', :unless => Puppet::Util::Platform.windows?  do
-    it 'returns the new puppetserver directory when no ca dir is present' do
-      expect(Puppet).to_not receive(:log_ca_migration_warning)
-      expect(Puppet.default_cadir).to eq('/etc/puppetlabs/puppetserver/ca')
-    end
-
-    it 'warns when a CA dir exists in the current ssldir' do
+    it 'defaults to the puppet/ssl dir when it detects a ca directory there' do
       cadir = File.join(Puppet[:ssldir], 'ca')
       FileUtils.mkdir_p(cadir)
-      expect(Puppet).to receive(:log_ca_migration_warning)
       expect(Puppet.default_cadir).to eq(cadir)
```

Comment:
In the other tests, we call `initialize_settings` and test `Puppet[:cadir]` rather than checking `default_cadir`. I don't know if one makes more sense than the other, but I think we should be consistent.

Reply:
I feel like this was a question on the original PR too and there was a reason it was left this way... does anyone else remember that?

Reply:
I don't really care either way, but there's a whole rigamarole in order to test that scenario; I deleted the usage of that pattern with this PR—because I think it kinda sucks and is very confusing— but I can put it back in if you want it to be consistent.

It would look something like this:
``` ruby
    it 'defaults to the puppet/ssl dir when it detects a ca directory there' do
      FileUtils.mkdir_p(Puppet[:confdir])
      cadir_location = File.join(Puppet[:ssldir], 'ca')
      # This is fun; in order to get the confdir setting from the spec_helper to
      # not be deleted by the #unsafe_clear in Puppet::Settings#parse_config,
      # we can code it into the confdir itself, and thus we can avoid getting
      # that setting wiped out during Puppet.initialize_settings.
      File.write(File.join(Puppet[:confdir], 'puppet.conf'),
                 "cadir = #{cadir_location}\n
                  confdir = #{Puppet[:confdir]}")
      Puppet.initialize_settings
      expect(Puppet[:cadir]).to eq(cadir_location)
    end
```

Reply:
ahh. Then can the other ones test `default_cadir`? Or would that mean something different?

Reply:
I think I'll just add the above code to the commit; the settings test are more useful than the method tests. Someday someone will refactor the spec_helper, but until then...